.circleci/config.yml

```diff
@@ -155,30 +155,53 @@ jobs:
             sudo service docker restart
 
       - run:
-          name: Bootstrap the test applications in OpenShift
+          name: Run local OpenShift cluster & configure environment
           command: |
             export OPENSHIFT_DOMAIN=$(hostname -I | awk '{print $1}')
             export K8S_AUTH_HOST="https://${OPENSHIFT_DOMAIN}:8443"
             oc cluster up --server-loglevel=5 --public-hostname="${OPENSHIFT_DOMAIN}"
             oc login "${K8S_AUTH_HOST}" -u developer -p developer --insecure-skip-tls-verify=true
             export K8S_AUTH_API_KEY="$(oc whoami -t)"
-            echo "arnold" | docker run --rm -i \
-                --env-file env.d/development \
-                --env K8S_AUTH_API_KEY \
-                --env K8S_AUTH_HOST \
-                --env OPENSHIFT_DOMAIN \
-                "arnold:$(tr -d '\n' < VERSION)" \
-                ansible-playbook bootstrap.yml --ask-vault-pass -e "env_type=ci"
-            for try in $(seq 5); do
-              echo "Testing hello application http response ($try)"
-              curl -vk "https://next.hello.ci-patient0.${OPENSHIFT_DOMAIN}.nip.io/" 2> /tmp/hello.err > /tmp/hello.out
-              if grep "HTTP/1.1 200 OK" /tmp/hello.err && grep "Hello OpenShift! by Arnold" /tmp/hello.out; then
-                exit 0
-              fi
-              sleep 10
-            done
-            exit 1
+            # Set OpenShift's environment variables for future steps
+            echo "export OPENSHIFT_DOMAIN=${OPENSHIFT_DOMAIN}" >> $BASH_ENV
+            echo "export K8S_AUTH_HOST=${K8S_AUTH_HOST}" >> $BASH_ENV
+            echo "export K8S_AUTH_API_KEY=${K8S_AUTH_API_KEY}" >> $BASH_ENV
+            source $BASH_ENV
 
+      - run:
+          name: Create applications bootstraping routine
+          command: |
+            cat << EOF > bin/ci-bootstrap
+              #!/usr/bin/env/bash
```

Comment:
Then this should be in bin/ as a script...

Reply:
I think we need much more comments on all this.

Reply:
I wanted to maintain a self-contained CI script. I'll add comments to explain what it does. In future optimizations, maybe we can extract this in a sugar script to make things simpler.